src/core/operations_problem_results.jl

```diff
@@ -158,9 +166,79 @@ function write_optimizer_stats(res::OperationsProblemResults, directory::Abstrac
     JSON.write(joinpath(directory, "optimizer_stats.json"), JSON.json(data))
 end
 
+function _read_realized_results(
+    result_values::Dict{Symbol, DataFrames.DataFrame},
+    names::Union{Nothing, Vector{Symbol}},
+)
+    existing_names = collect(keys(result_values))
+    names = isnothing(names) ? existing_names : names
+    _validate_names(existing_names, names)
+    return filter(p -> (p.first ∈ names), result_values)
+end
+
+function _read_results(
+    result_values::Dict{Symbol, DataFrames.DataFrame},
+    names::Union{Nothing, Vector{Symbol}},
+    initial_time::Dates.DateTime,
+)
+    realized_results = _read_realized_results(result_values, names)
+    results = FieldResultsByTime()
+    for (name, df) in realized_results
+        results[name] = ResultsByTime(initial_time => df)
+    end
+    return results
+end
+
+function read_realized_variables(
+    res::OperationsProblemResults;
+    names::Union{Vector{Symbol}, Nothing} = nothing,
+)
+    variable_values = get_variables(res)
+    return _read_realized_results(variable_values, names)
+end
+
+function read_realized_parameters(
+    res::OperationsProblemResults;
+    names::Union{Vector{Symbol}, Nothing} = nothing,
+)
+    parameter_values = IS.get_parameters(res)
+    return _read_realized_results(parameter_values, names)
+end
+
+function read_realized_duals(
+    res::OperationsProblemResults;
+    names::Union{Vector{Symbol}, Nothing} = nothing,
+)
+    dual_values = get_duals(res)
+    return _read_realized_results(dual_values, names)
+end
+
 function read_variables(
-    problem::OperationsProblem,
+    res::OperationsProblemResults;
+    names::Union{Vector{Symbol}, Nothing} = nothing,
+)
+    result_values = get_variables(res)
+    return _read_results(result_values, names, first(get_timestamps(res)))
+end
+
+function read_parameters(
+    res::OperationsProblemResults;
     names::Union{Vector{Symbol}, Nothing} = nothing,
 )
-    variables = get_variables(problem)
+    result_values = IS.get_parameters(res)
+    return _read_results(result_values, names, first(get_timestamps(res)))
+end
+
+function read_duals(
+    res::OperationsProblemResults;
+    names::Union{Vector{Symbol}, Nothing} = nothing,
+)
+    result_values = get_duals(res)
+    return _read_results(result_values, names, first(get_timestamps(res)))
+end
+
+function read_optimizer_stats(res::OperationsProblemResults)
```

Comment:
I started down this path in my PR to refactor these results because I understood that users might want consistency. However, I decided against including the `read_X` functions because it didn't seem useful. Getting back a DataFrame that can only have a single entry is less useful that just returning that single entry. If you have an OperationsProblemResults and want the optimizer stats, you should call `get_optimizer_stats()`. It appears that I included a partial `read_variables` function. That was inadvertent.

There was already existing code doing things like `res.variable_values[:P__ThermalStandard]`. I thought that was more convenient than having to do `first(values(read_variables(res)[:P__ThermalStandard]))`.

Just explaining why I didn't do this. I don't disagree with the notion of having consistent interfaces.